src/token.rs

```diff
@@ -160,6 +160,38 @@ impl Handles {
     }
 }
 
+/* Sow = Silly Object Wrapper */
+struct Sow<'a, T> {
+    ro: Option<&'a T>,
+    vo: Option<T>,
```

Comment:
I think these would be better to have either comment or expanded names so the code would be readable. I understand that they are likely abbreviation for `ref object` and `value object`, but it might not be obvious on first sight. And given that they are internal, shortening them like this wont save users any time writing the names.

Reply:
This was just stop gap, I asked on our rust channel and they suggested to use Cow, which does exactly what I need. I should have known, I used Cow previously, but somehow did not realize it could be easily used this way.
Replaced with simpler code